src/Adapters/Files/FilesAdapter.js

```diff
@@ -31,12 +31,14 @@ export class FilesAdapter {
    * @discussion the contentType can be undefined if the controller was not able to determine it
    * @param {object} options - (Optional) options to be passed to file adapter (S3 File Adapter Only)
    * - tags: object containing key value pairs that will be stored with file
-   * - metadata: object containing key value pairs that will be sotred with file (https://docs.aws.amazon.com/AmazonS3/latest/user-guide/add-object-metadata.html)
-   * @discussion options are not supported by all file adapters. Check the your adapter's documentation for compatibility
+   * - metadata: object containing key value pairs that will be stored with file (https://docs.aws.amazon.com/AmazonS3/latest/user-guide/add-object-metadata.html)
+   * @discussion options are not supported by all file adapters. Check the your adapter's documentation for compatibility 
+   * @param {Config} config -(Optional) server configuration
+   * @discussion config is not supported by all file adapters. Check the your adapter's documentation for compatibility
    *
-   * @return {Promise} a promise that should fail if the storage didn't succeed
+   * @return {Promise<any>|Promise<{url?: string, name?: string, location?: string}>} Either a plain promise that should fail if storage didn't succeed, or a promise resolving to an object containing url and/or an updated filename and/or location (if relevant)
```

Comment:
From the description I believe the notation would be `{Promise<{url?: string, name?: string, location?: string}|undefined>}`, i.e. a promise that resolves with ... or rejects with `undefined`?

Reply:
I switched it to  {Promise<{url?: string, name?: string, location?: string}>|Promise<undefined>}